lib/modules/manager/cargo/extract.ts

```diff
@@ -184,11 +189,27 @@ export async function extractPackageFile(
       targetDeps = targetDeps.concat(deps);
     });
   }
+
+  const workspaceSection = cargoManifest.workspace;
+  let workspaceDeps: PackageDependency[] = [];
+  if (workspaceSection) {
```

Comment:
i don't think we need the condition here. the extract function should already handle it

Reply:
workspaceSection is either a CargoSection or undefined.

And extractFromSection requires a defined CargoSection. ie.

```typescript
const sectionContent = parsedContent[section];
```

Af far as i recall an undefined map will throw if index accessed.